smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaSource.java

```diff
@@ -60,14 +60,17 @@
 
         Objects.requireNonNull(topic, "The topic must be set, or the name must be set");
 
-        Flowable<KafkaConsumerRecord<K, V>> flowable = consumer.toFlowable();
+        Flowable<KafkaConsumerRecord<K, V>> flowable = consumer.toFlowable()
+                .doOnError(t -> LOGGER.error("Unable to read a record from Kafka topic '{}'", topic, t));
 
         if (config.getOptionalValue("retry", Boolean.class).orElse(true)) {
             Integer max = config.getOptionalValue("retry-attempts", Integer.class).orElse(5);
             flowable = flowable
-                    .retryWhen(attempts -> attempts
-                            .zipWith(Flowable.range(1, max), (n, i) -> i)
-                            .flatMap(i -> Flowable.timer(i, TimeUnit.SECONDS)));
+                    .retryWhen(attempts -> {
+                        return attempts
```

Comment:
Not huge, but is the use of `return` needed, appears to be a single line statement? Is the code change actually changing anything?